src/hooks/api/useProposalsStatusChangeUser.js

```diff
@@ -0,0 +1,101 @@
+import { useState } from "react";
+import * as act from "src/actions";
+import * as sel from "src/selectors";
+import { useAction, useSelector } from "src/redux";
+import head from "lodash/head";
+import isEmpty from "lodash/fp/isEmpty";
+import uniq from "lodash/uniq";
+import compact from "lodash/compact";
+import useFetchMachine, {
+  VERIFY,
+  FETCH,
+  REJECT,
+  RESOLVE,
+  START
+} from "src/hooks/utils/useFetchMachine";
+import { shortRecordToken } from "src/helpers";
+
+// useProposalsStatusChangeUser receives a key-value object of proposals
+// and a status. For all proposals that are in the passed in status, the
+// hook adds a 'statuschangeusername' field to the proposal object, which
+// corresponds to the username of the user that submitted the status change
+// action to the server.
+export default function useProposalsStatusChangeUser(proposals = {}, status) {
+  const [publicKeys, setPublicKeys] = useState([]);
+  const onSearchUser = useAction(act.onSearchUser);
+  const resultsByPk = useSelector(sel.queryResultsByPublicKey);
+  const resultsByID = useSelector(sel.searchResultsByID);
+
+  // The public keys for the users we need to search for
+  const unfetchedPublicKeys = uniq(
+    Object.values(proposals).flatMap((prop) =>
+      prop?.status === status ? prop?.statuschangepk : []
+    )
+  ).filter((pk) => pk && !resultsByPk[pk]);
```

Comment:
```suggestion
  const unfetchedPublicKeys = flow(
    values,
    flatMap((prop) => (prop.status === status ? prop.statuschangepk : [])),
    uniq,
    filter((pk) => pk && !resultsByPk[pk])
  )(proposals);
```

I think you should go for a functional approach here

Reply:
It is using a functional composition approach, but yeah, I suppose we can use lodash's `flow` here to improve readability

Reply:
it's a matter of taste for me. Both will work!

Reply:
checked and we can't use flow in this situation because `flatMap` and `filter` are native prototype Array methods, so they need to be called by an object of type Array, or we get `flatMap` and `filter` are not defined.